### PR TITLE
build: restrucutre github workflows

### PR DESCRIPTION
**Pull request by SanderBlom**
# Changed to Terraform version 1.8 and trigger e2e tests using comments

## Description
All the tests should now run using terraform 1.9 
The end to end tests now takes about 20 minutes. As we dont need to do this for every new commit to the PR I propose that we can trigger this using a comment when needed. After the new changes it should now only run if you write the comment `/start_e2e` in the pull request or when creating a new release.

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Documentation update
- [x] Other

## Testing
See the testing [docs](../tests/readme.md) for more information about how to test your code 
- [ ] Performed a local `terraform plan`, `apply`, and `destroy` to validate changes.
- [ ] All tests are passing.
- [ ] Added, updated, or removed tests when appropriate

## Checklist

Before submitting this PR, please ensure the following:

- [x] My commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) style guide
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings. I have ran `terraform validate` in the root module and all sub modules where I have made changes

## Additional Information

Provide any additional information or context about the pull request here.

---

Thank you for contributing to Station!

### CODE REVIEW

**Comment by SanderBlom** [comment]
this is a test

File: .github/workflows/terraform.yaml [hunk 1]
```diff
@@ -93,9 +95,50 @@ jobs:
...
 
   apply_and_destroy:
-    if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch' || github.event_name == 'release' || (github.event_name == 'push' && github.ref == 'refs/heads/trunk')
+    if: >
+      (
+        github.event_name == 'workflow_dispatch' ||
+        github.event_name == 'release'
+      ) || (
+        github.event.issue.pull_request &&
+        contains(github.event.comment.body, '/start_e2e')
```
Comment (kimfy) [inline .github/workflows/terraform.yaml]: I suggest we change the expression to `==` and evaluate who the caller is (me and you fex.). You can use `github.event.actor`: https://docs.github.com/en/rest/using-the-rest-api/github-event-types?apiVersion=2022-11-28.

```shell
github.event.issue.pull_request && github.event.comment.body == '/start_e2e'
```

**Review by kimfy — APPROVED** [review]

**Pull request merged**